stdlib/public/core/Runtime.swift.gyb

```diff
@@ -408,37 +408,17 @@ internal func _float${bits}ToString(
   _ value: Float${bits}, debug: Bool
 ) -> String {
 
-  if !value.isFinite {
-    let significand = value.significandBitPattern
-    if significand == 0 {
-      // Infinity
+  if value.isInfinite {
       return value.sign == .minus ? "-inf" : "inf"
+  } else {
+    _sanityCheck(MemoryLayout<_Buffer32>.size == 32)
+    var buffer = _Buffer32()
+    return buffer.withBytes { (bufferPtr) in
+      let actualLength = _float${bits}ToStringImpl(bufferPtr, 32, value, debug)
+      return String._fromWellFormedCodeUnitSequence(
+                 UTF8.self,
+                 input: UnsafeBufferPointer(start: bufferPtr, count: Int(actualLength)))
```

Comment:
Does anything need to change here to take advantage of @milseman's recent Small String work?

Reply:
Small string support hasn't been merged and I haven't come up with a clean alternative to `_fromWellFormedCodeUnitSequence` for them yet. When you merge, file a bug against me

Reply:
Ideally, I'd like to have something like `String._unsafeFromWellFormedASCII(buffer:)` to use here.  If the buffer length is short, build and return a small ASCII string.  Otherwise, allocate and `memcpy` to build a long ASCII string.  In either case, omitting the charset shenanigans would be a sizable win.

Reply:
I have just what you need in the small-string branch: https://github.com/milseman/swift/blob/so_smol/stdlib/public/core/String.swift#L647